### PR TITLE
xds_override_host LB: move XdsOverrideHostLbConfig to `internal` namespace

### PR DESCRIPTION
**Pull request by eugeneo**

### CODE REVIEW

File: test/core/client_channel/lb_policy/xds_override_host_lb_config_parser_test.cc [hunk 1]
```diff
@@ -45,6 +47,43 @@ TEST(XdsOverrideHostConfigParsingTest, ValidConfig) {
       ServiceConfigImpl::Create(ChannelArgs(), service_config_json);
   ASSERT_TRUE(service_config.ok());
   EXPECT_NE(*service_config, nullptr);
+  auto global_config = static_cast<internal::ClientChannelGlobalParsedConfig*>(
+      (*service_config)->GetGlobalParsedConfig(0));
```
[the remark this answers is not in the document]
Comment (eugeneo) [reply test/core/client_channel/lb_policy/xds_override_host_lb_config_parser_test.cc]: Done.

File: src/core/ext/filters/client_channel/lb_policy/xds/xds_override_host.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+#include "src/core/lib/gprpp/ref_counted_ptr.h"
+#include "src/core/lib/gprpp/validation_errors.h"
+#include "src/core/lib/json/json.h"
+#include "src/core/lib/json/json_args.h"
+#include "src/core/lib/json/json_object_loader.h"
+#include "src/core/lib/load_balancing/lb_policy.h"
+
+namespace grpc_core {
+namespace internal {
```
[the remark this answers is not in the document]
Comment (eugeneo) [reply src/core/ext/filters/client_channel/lb_policy/xds/xds_override_host.h]: Done

File: src/core/ext/filters/client_channel/lb_policy/xds/xds_override_host.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+constexpr absl::string_view kXdsOverrideHost = "xds_override_host_experimental";
```
[the remark this answers is not in the document]
Comment (eugeneo) [reply src/core/ext/filters/client_channel/lb_policy/xds/xds_override_host.h]: Done

**Review by eugeneo — COMMENTED** [review]
Thank you for the review!

**Review by markdroth — APPROVED** [review]
Looks great!

**Pull request merged**